File: addons/FoundationSteamworks/src/private/SteamInitialisationResponse.cpp
#include "SteamInitialisationResponse.h"

void SteamInitialisationResponse::_bind_methods()
{
    ClassDB::bind_method(D_METHOD("GetSuccess"), &SteamInitialisationResponse::GetSuccess);
    ClassDB::bind_method(D_METHOD("SetSuccess", "v"), &SteamInitialisationResponse::SetSuccess);
    ADD_PROPERTY(PropertyInfo(Variant::BOOL, "Success"), "SetSuccess", "GetSuccess");

    ClassDB::bind_method(D_METHOD("GetShouldRestart"), &SteamInitialisationResponse::GetShouldRestart);
    ClassDB::bind_method(D_METHOD("SetShouldRestart", "v"), &SteamInitialisationResponse::SetShouldRestart);
    ADD_PROPERTY(PropertyInfo(Variant::BOOL, "ShouldRestart"), "SetShouldRestart", "GetShouldRestart");

    ClassDB::bind_method(D_METHOD("GetMessage"), &SteamInitialisationResponse::GetMessage);
    ClassDB::bind_method(D_METHOD("SetMessage", "v"), &SteamInitialisationResponse::SetMessage);
    ADD_PROPERTY(PropertyInfo(Variant::STRING, "Message"), "SetMessage", "GetMessage");
}

File: Archive/addons/FoundationSteamworks/CSharp/Data/SteamInitialisationResponse.cs
using Godot;

namespace Heathen.SteamworksIntegration
{
    public class SteamInitialisationResponse
    {
        private GodotObject _instance;

        public SteamInitialisationResponse(GodotObject instance) => _instance = instance;

        public bool Success => (bool)_instance.Get("Success");

        public bool ShouldRestart => (bool)_instance.Get("ShouldRestart");

        public string Message => (string)_instance.Get("Message");
    }
}
